src/lib/swapper/swappers/ThorchainSwapper/utils/makeMemoWithShortenedFinalAssetAmount.ts

```diff
@@ -0,0 +1,91 @@
+import assert from 'assert'
+import { toBaseUnit } from 'lib/math'
+import { assertAndProcessMemo } from 'lib/utils/thorchain/memo'
+
+import { AGGREGATOR_EXPONENT_LENGTH, THORCHAIN_PARSER_MAXIMUM_PRECISION } from '../constants'
+import { MEMO_PART_DELIMITER } from './constants'
+
+// Allows us to convert a THORChain parser value back to its base unit
+// for testing purposes
+export const thorchainParserToBaseUnit = (thorchainParserValue: string) => {
+  if (thorchainParserValue.length < 3) {
+    throw new Error('Number too short to contain an exponent.')
+  }
+
+  const exponent = parseInt(thorchainParserValue.slice(-2), 10)
+
+  const baseNumber = thorchainParserValue.slice(0, -2)
+
+  const fullNumber = toBaseUnit(baseNumber, exponent === 1 ? 0 : exponent)
+
+  return fullNumber
+}
```

Comment:
This is unused

Reply:
Removed it for now, let's implement it in a follow up PR if needed but it would require a bit more testing as it's a brand new util that we should maintain, for now I added some comments to convert the amounts back and understand things